llvm/lib/IR/LegacyPassManager.cpp

```diff
@@ -1535,7 +1535,11 @@ MPPassManager::runOnModule(Module &M) {
   if (EmitICRemark)
     InstrCount = initSizeRemarkInfo(M, FunctionToInstrCount);
 
-  for (unsigned Index = 0; Index < getNumContainedPasses(); ++Index) {
+  // EVM local begin
+  for (unsigned Index = 0;
+       Index < getNumContainedPasses() && !M.getContext().getSpillAreaSize();
```

Comment:
Do we need this condition to be a part of the loop?

Reply:
Yes, because the return value of getSpillAreaSize() may change after the last pass has been executed.
But we can put it at the end of the loop body:
```
if (M.getContext().getSpillAreaSize())
      break;
```

Reply:
Yes, it would be better to not to modify the condition and do as you suggested. Adding necessary documentation there.

Reply:
Sure, done.